File: ios/Plugin/CapacitorFlashPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(CapacitorFlashPlugin, "CapacitorFlash",
           CAP_PLUGIN_METHOD(isAvailable, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(switchOn, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(switchOff, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(isSwitchedOn, CAPPluginReturnPromise);
)

File: ios/Plugin/CapacitorFlashPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(CapacitorFlashPlugin)
public class CapacitorFlashPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "CapacitorFlashPlugin"
    public let jsName = "CapacitorFlash"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "isAvailable", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "switchOn", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "switchOff", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "isSwitchedOn", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "toggle", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = CapacitorFlash()

    @objc func isAvailable(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isAvailable()
        ])
    }

    @objc func switchOn(_ call: CAPPluginCall) {
        let intensity = call.getFloat("intensity") ?? 1.0
        call.resolve([
            "value": implementation.switchOn(intensity: intensity)
        ])
    }

    @objc func switchOff(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.switchOff()
        ])
    }

    @objc func isSwitchedOn(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.isSwitchedOn()
        ])
    }

    @objc func toggle(_ call: CAPPluginCall) {
        call.resolve([
            "value": implementation.toggle()
        ])
    }
}
